src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/Stash.java

```diff
@@ -0,0 +1,61 @@
+package com.minecolonies.coremod.colony.buildings.workerbuildings;
+
+import com.google.common.collect.ImmutableCollection;
+import com.google.common.collect.ImmutableList;
+import com.minecolonies.api.colony.IColony;
+import com.minecolonies.api.colony.IColonyView;
+import com.minecolonies.api.colony.buildings.ModBuildings;
+import com.minecolonies.api.colony.buildings.registry.BuildingEntry;
+import com.minecolonies.api.colony.requestsystem.resolver.IRequestResolver;
+import com.minecolonies.coremod.colony.buildings.AbstractBuilding;
+import com.minecolonies.coremod.colony.buildings.views.AbstractBuildingView;
+import net.minecraft.util.math.BlockPos;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Class used to manage the stash building block.
+ */
+public class Stash extends AbstractBuilding
+{
```

Comment:
I know this is a bit out of the scope of this PR. But maybe you're up to it. Maybe it makes sense to extract what Stash and PostBox need from AbstractBuilding into a separate abstract super class and have Stash and PostBox extend those.

So that we don't need the "!(building instanceof PostBox) && !(building instanceof Stash)" checks in a row, that might be cleaner.

Or at least an IRSComponent interface that both Stash and PostBox implement and we check only if !instanceof IRSComponent

Reply:
I thought about that as well, but I figured it was a bit out of scope of the PR. Also, it wouldn't need an extraction only from `AbsctractBuilding`, but the whole hierarchy, as there's a lot things there that isn't needed whatsoever. And I actually thought of, maybe, letting Stash and PostBox be also built along with an schematic? I think it's a whole project on it's on right there, and being new to the project (and to Minecraft modding), I'm not confident enough to pull drastic changes such as these.

But the idea of the interface is actually good. The IRSComponent you thought would be just an empty interface?

Reply:
Yes, I agree. Yes it would just be an empty interface.